src/CommunityToolkit.Maui.UnitTests/Views/UniformGrid_Tests.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using CommunityToolkit.Maui.UnitTests.Mocks;
+using CommunityToolkit.Maui.Views;
+using Microsoft.Maui;
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Graphics;
+using NSubstitute;
+using Xunit;
+
+namespace CommunityToolkit.Maui.UnitTests.Views;
+
+public class UniformGridTests : BaseTest
+{
+	const double _childCount = 3;
+	const double _childWidth = 20;
+	const double _childHeight = 10;
+
+	readonly UniformGrid _uniformGrid;
+	readonly IView _uniformChild;
+
+	public UniformGridTests()
+	{
+		_uniformChild = Substitute.For<IView>();
+
+		_uniformGrid = new UniformGrid()
+		{
+			Children =
+			{
+				_uniformChild,
+				_uniformChild,
+				_uniformChild
+			}
+		};
+	}
+
+	[Fact]
+	public void MeasureUniformGrid_NoWrap()
+	{
+		var expectedSize = new Size(_childWidth * _childCount, _childHeight);
+		var childSize = new Size(_childWidth, _childHeight);
+
+		SetupChildrenSize(childSize);
+
+		var actualSize = _uniformGrid.Measure(_childWidth * _childCount, _childHeight * _childCount);
+
+		Assert.Equal(expectedSize, actualSize);
+	}
+
+	[Fact]
+	public void MeasureUniformGrid_WrapOnNextRow()
+	{
+		var expectedSize = new Size(_childWidth, _childHeight * _childCount);
+		var childSize = new Size(_childWidth, _childHeight);
+
+		SetupChildrenSize(childSize);
+
+		var actualSize = _uniformGrid.Measure(_childWidth, _childHeight * _childCount);
+
+		Assert.Equal(expectedSize, actualSize);
+	}
+
+	[Fact]
+	public void ArrangeChildrenUniformGrid()
+	{
+		var expectedSize = new Size(_childWidth, _childHeight);
+
+		SetupChildrenSize(expectedSize);
+
+		var actualSize = _uniformGrid.ArrangeChildren(new Rectangle(0, 0, _childWidth * _childCount, _childHeight * _childCount));
+
+		Assert.Equal(expectedSize, actualSize);
+	}
+
+	[Fact]
+	public void MaxRowsArrangeChildrenUniformGrid()
+	{
+		var expectedSize = new Size(_childWidth, _childHeight);
+
+		SetupChildrenSize(expectedSize);
+
+		_uniformGrid.MaxColumns = 1;
+		_uniformGrid.MaxRows = 1;
+
+		var actualSize = _uniformGrid.Measure(double.PositiveInfinity, double.PositiveInfinity);
+
+		Assert.Equal(expectedSize, actualSize);
+	}
+
+	void SetupChildrenSize(Size size)
+	{
+		_uniformChild.Measure(double.PositiveInfinity, double.PositiveInfinity).ReturnsForAnyArgs(size);
```

Comment:
Can we accomplish this Unit Test without NSubstitute?

I'd prefer that we test against the actual public APIs and not use tools like NSubstitute to mock/inject values.

Reply:
I tried the actual API and it returns 0 for the real size

Reply:
Sounds like you've either found a bug, or we need to "Render" the element. 

Perhaps you could mock something up for the rendering, similar to how we created the [`.EnableAnimations()` extension](https://github.com/CommunityToolkit/Maui/blob/0b992330d8c109863b55ffe193bff1415c6bc905/src/CommunityToolkit.Maui.UnitTests/Mocks/AnimationExtensions.cs#L14) method that allows us to animate an `IView`.

Reply:
I am not sure if it is possible. .NET MAUI team also uses NSubstiture for Measure tests: https://github.com/dotnet/maui/blob/main/src/Controls/tests/Core.UnitTests/Layouts/LayoutCompatTests.cs

Reply:
well, I found a way to remove NSubstitute dependency. I will update PR in a few minutes